orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -966,6 +966,43 @@ def run_step(model, input):
         _test_helpers.assert_values_are_close(ort_prediction, pt_prediction)
         _test_helpers.assert_values_are_close(ort_input.grad, pt_input.grad)
 
+# Since multinomial is a generator function, we do not have to test for gradient
+# Two consecutive calls on the torch.multinomail on a probability distribution with more 
+# than one index with non-zero probability(eg, [0, 10, 3, 0]) will not always result in 
+# the same output. Output could be [1] or [2] when sampling one index.
+# Hence we test this with an input that will always give deterministic output 
+# with only one index having non-zero probability.
+@pytest.mark.parametrize("input", ([0, 10, 0, 0], [[0, 10, 0, 0], [0, 10, 0, 0]]))
+@pytest.mark.parametrize("num_samples, replacement", ((1, False), (2, True)))
+def test_aten_multinomial(input, num_samples, replacement):
+    class NeuralNetDiagonal(torch.nn.Module):
+        def __init__(self, num_samples, replacement):
+            super(NeuralNetDiagonal, self).__init__()
+            self.num_samples = num_samples
+            self.replacement = replacement
+
+        def forward(self, input):
+            return torch.multinomial(input, self.num_samples, self.replacement)
+
+    torch.manual_seed(5032)
+    torch.backends.cudnn.deterministic = True
+    torch.backends.cudnn.benchmark = False
```

Comment:
is torch.backends.cudnn.benchmark related to multinomial kernel?

Reply:
It shouldn't be required for this test. removing